fendermint/testing/graph-test/Makefile.toml

```diff
@@ -37,17 +38,62 @@ cargo run -p fendermint_eth_api --release --example greeter -- \
 
 [tasks.graph-setup]
 script = """
-echo "TODO: configure the graph"
+set -x
+echo "Start configuring subgraph..."
+echo "Found deployed contract:"
 cat ${TEST_DATA_DIR}/greeter.json
-echo "TODO: docker-compose up the graph"
+echo "\n"
+
+cd ${CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY}/fendermint
+cp -r testing/graph-test/subgraph testing/graph-test/test-data
+mkdir testing/graph-test/test-data/subgraph/config
+mkdir testing/graph-test/test-data/subgraph/abis
+cp testing/graph-test/test-data/greeter.json testing/graph-test/test-data/subgraph/config/testnet.json
+cp testing/contracts/Greeter.abi testing/graph-test/test-data/subgraph/abis/IGreeter.json
+cd testing/graph-test/test-data/subgraph
+
+npm install
+npm run compile
+npm run graph-node
+
+echo "Waiting for Graph node to start..."
+# Notice that subgraph node does takes a while to spin up
+sleep 30s
+
+npm run create-local
+npm run deploy-local
+
+echo "Subgraph is running now."
 """
 
 [tasks.graph-test]
 script = """
-echo "TODO: run GraphQuery with curl"
+echo "Query against local subgraph server using GraphQL."
+GREETING_MSG_0=$(curl -s 'http://localhost:8000/subgraphs/name/Greeter' --data-raw '{"query":"{\\n  greetings(sort: \\"createdAt:desc\\") {\\n    id\\n    currentGreeting\\n  }\\n}","variables":null,"extensions":{"headers":null}}' | jq -r '.data.greetings[0].currentGreeting')
+GREETING_MSG_1=$(curl -s 'http://localhost:8000/subgraphs/name/Greeter' --data-raw '{"query":"{\\n  greetings(sort: \\"createdAt:desc\\") {\\n    id\\n    currentGreeting\\n  }\\n}","variables":null,"extensions":{"headers":null}}' | jq -r '.data.greetings[1].currentGreeting')
+TARGET_MSGS=' [ "Welcome, weary traveller!", "Howdy doody!" ] '
+echo $GREETING_MSG_0
+echo $GREETING_MSG_1
+
+# It's hard to predict which message get indexed first because they always belong to the same block.
```

Comment:
How odd; you'd think that because the deployment waits for the receipt to get the contract address, the first event will be in an earlier block :thinking: